src/scheduler/workers/processListsBeforeAndDuringStart/main.ts

```diff
@@ -95,42 +94,34 @@ async function processProviderEmailsForListItems(list: List, listItems: ListItem
 
     // get the most recent audit record to determine if the email has already been sent for the start milestone
     if (annualReviewRef) {
-      const { result: events } = await findAuditEvents(annualReviewRef, "REMINDER", "listItem", listItem.id);
+      const { result: events } = await findAllReminderEvents(annualReviewRef, listItem.id);
       if (events?.length) {
-        const audit = events.pop();
-        isEmailSent = isEmailSentBefore(audit as Audit, "sendStartedProviderEmail");
+        const event = events.pop();
+        isEmailSent = isEmailSentBefore(event, "sendStartedProviderEmail");
       }
     }
 
     // email the provider and add an audit record
     if (!isEmailSent) {
-      const providerEmailResult = await emailProvider(list, listItem);
+      const { result } = await sendAnnualReviewStartEmail(list, listItem);
       const annualReviewRef = list.jsonData.currentAnnualReview?.reference;
 
-      if (providerEmailResult.result) {
-        await recordListItemEvent(
-          {
-            eventName: "reminder",
-            itemId: listItem.id,
-            annualReviewRef,
-            reminderType: "sendStartedProviderEmail",
-          },
-          AuditEvent.REMINDER,
-          "listItem",
-          "scheduler"
-        );
+      if (result) {
+        await addReminderEvent(listItem.id, result as SendEmailResponse, ["sendStartedProviderEmail"], annualReviewRef);
       }
     }
   }
 }
 
-async function getLatestReminderAuditEvent(annualReviewRef: string, auditType: "user" | "list" | "listItem") {
-  const { result: events } = await findAuditEvents(annualReviewRef, "REMINDER", auditType);
-  let audit: Audit | undefined;
+async function getLatestReminderEvent(
+  callback: () => Promise<{ result?: Event[]; error?: Error } | { result?: Audit[]; error?: Error }>
```

Comment:
Callbacks should take arguments. This pattern is fine but needs to be unwrapped 

```.ts
// describe the function you want to run in callback - in this case we want it to take a parameter of type string, and return the result/error thing.
type GetLatestReminderCallback = (annualReviewRef: string) => Promise<{ result?: Event[] | Audit[]; error?: Error }>;

async function getLatestReminderEvent(annualReviewRef: string, callback: GetLatestReminderCallback) {
  await callback(annualReviewRef);
}

const audit = getLatestReminderEvent("abcdef", findReminderAuditEvents);
const listItemAudit = getLatestReminderEvent("abcdef", findReminderAllReminderEvents);
```

Reply:
This is good to now. Do you have any documentation to support this. Just for my learning?